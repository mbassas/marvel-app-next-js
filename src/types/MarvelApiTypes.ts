export interface DataWrapper<T> {
  code: string;
  status: string;
  copyright: string;
  attributionText: string;
  attributionHTML?: string;
  data: DataContainer<T>;
  etag?: string;
}

export interface DataContainer<T> {
  offset: number;
  limit: number;
  total: number;
  count: number;
  results: Array<T>;
}

export type CharacterDataWrapper = DataWrapper<Character>;
export type CharacterDataContainer = DataContainer<Character>;

export interface Character {
  id: number;
  name: string;
  description: string;
  modified: Date;
  resourceURI: string;
  urls: Url[];
  thumbnail: Image;
  comics: ComicList;
}

export interface Url {
  type: string;
  url: string;
}

export interface Image {
  path: string;
  extension: string;
}

export interface ComicList {
  available: number;
  returned: number;
  collectionURI: string;
  items: ComicSummary[];
}

export interface ComicSummary {
  resourceURI: string;
  name: string;
}

export type ComicDataWrapper = DataWrapper<Comic>;
export type ComicDataContainer = DataContainer<Comic>;

export interface Comic {
  id: number;
  title: string;
  dates: ComicDate[];
  thumbnail: Image;
}

interface ComicDate {
  type: string;
  date: Date;
}
