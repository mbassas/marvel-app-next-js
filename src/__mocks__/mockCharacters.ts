import { Character } from "@/types/MarvelApiTypes";

export const mockCharacters: Pick<
  Character,
  "name" | "thumbnail" | "id" | "description"
>[] = [
  {
    id: 1,
    name: "Test Character",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 2,
    name: "Test Character 2",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 3,
    name: "Test Character 3",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 4,
    name: "Test Character 4",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 5,
    name: "Test Character 5",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 6,
    name: "Test Character 6",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 7,
    name: "Test Character 7",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 8,
    name: "Test Character 8",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 9,
    name: "Test Character 9",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
  {
    id: 10,
    name: "Test Character 10",
    thumbnail: {
      path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
      extension: "jpg",
    },
    description: "Test description",
  },
];
