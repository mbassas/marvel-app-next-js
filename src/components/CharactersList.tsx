"use client";
import { Character } from "@/types/MarvelApiTypes";
import CharacterCard from "./CharacterCard";
import styled from "styled-components";

const CharactersList: React.FC<{ characters: Character[] }> = ({
  characters,
}) => {
  return (
    <Ul>
      {characters.map((character) => (
        <Li key={character.id}>
          <CharacterCard
            name={character.name}
            thumbnail={character.thumbnail}
          />
        </Li>
      ))}
    </Ul>
  );
};

const Li = styled.li`
  list-style: none;
`;
const Ul = styled.ul`
  display: grid;
  grid-template-columns: repeat(auto-fill, 173px);
  row-gap: 16px;
  justify-content: space-between;
`;

export default CharactersList;
