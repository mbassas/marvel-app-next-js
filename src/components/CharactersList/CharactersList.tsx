"use client";
import { Character } from "@/types/MarvelApiTypes";
import CharacterCard from "../CharacterCard/CharacterCard";
import styled from "styled-components";
import Link from "next/link";

const CharactersList: React.FC<{ characters: Character[] }> = ({
  characters,
}) => {
  return (
    <>
      <Ul>
        {characters.map((character) => (
          <Li key={character.id}>
            <Link href={`/characters/${character.id}`}>
              <CharacterCard
                name={character.name}
                thumbnail={character.thumbnail}
                id={character.id}
                description={character.description}
              />
            </Link>
          </Li>
        ))}
      </Ul>
    </>
  );
};

const Li = styled.li`
  list-style: none;
`;
const Ul = styled.ul`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(172px, 1fr));
  gap: 16px;
  justify-content: space-between;
`;

export default CharactersList;
