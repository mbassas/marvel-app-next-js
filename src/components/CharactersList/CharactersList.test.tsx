import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CharactersList from "@/components/CharactersList/CharactersList";
import { Character } from "@/types/MarvelApiTypes";
import { mockCharacters } from "@/__mocks__/mockCharacters";

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useRouter: jest.fn().mockReturnValue({ refresh: jest.fn() }),
}));

describe("CharactersList test suite", () => {
  it("should render characters list", async () => {
    render(<CharactersList characters={mockCharacters} />);
    expect(await screen.findAllByRole("listitem")).toHaveLength(
      mockCharacters.length,
    );
  });

  it("should render characters list with no characters", async () => {
    render(<CharactersList characters={[]} />);
    expect(screen.getByRole("list")).toBeEmptyDOMElement();
  });

  it("should match snapshot", async () => {
    const { container } = render(
      <CharactersList characters={mockCharacters} />,
    );
    expect(container).toMatchSnapshot();
  });
});
