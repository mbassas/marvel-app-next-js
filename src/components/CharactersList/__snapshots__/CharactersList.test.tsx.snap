// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersList test suite should match snapshot 1`] = `
<div>
  <ul
    class="sc-52e6a265-1 bXWJfi"
  >
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
    <li
      class="sc-52e6a265-0 hlwSxP"
    >
      <a
        href="/characters/1"
      >
        <div
          class="sc-f4478397-2 clVUFy"
        >
          <div
            class="sc-f4478397-3 fQputf"
          >
            <div
              class="sc-f4478397-4 bWRWYe"
            >
              <img
                alt="Test Character 1"
                class="sc-f4478397-5 gKtkFz"
                loading="lazy"
                src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
              />
            </div>
            <div
              class="sc-f4478397-9 bPdtYJ"
            >
              <div
                class="sc-f4478397-6 fAqBFZ"
              >
                <div
                  class="sc-f4478397-0 evpbLf"
                />
                <p
                  class="sc-2db796cf-0 dOeWdF sc-f4478397-7 cvXNwo"
                >
                  Test Character 1
                </p>
                <button
                  aria-label="heart-icon"
                  class="sc-f4478397-1 fcjSIG"
                >
                  <svg
                    data-testid="heart-icon-outline"
                    fill="none"
                    height="11"
                    viewBox="0 0 26 25"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M7 1.67337L7.51885 0.818504L7 0.503596L6.48115 0.818505L7 1.67337ZM13 5.31499L12.4812 6.16985L13 6.48476L13.5188 6.16985L13 5.31499ZM1 5.31499L0.481151 4.46012L0 4.75215V5.31499H1ZM1 13.1185H0V13.58L0.351203 13.8794L1 13.1185ZM13 23.3497L12.3512 24.1106L13 24.6638L13.6488 24.1106L13 23.3497ZM25 13.1185L25.6488 13.8794L26 13.58V13.1185H25ZM25 5.31499H26V4.75215L25.5188 4.46012L25 5.31499ZM19 1.67337L19.5189 0.818505L19 0.503596L18.4812 0.818504L19 1.67337ZM6.48115 2.52824L12.4812 6.16985L13.5188 4.46012L7.51885 0.818504L6.48115 2.52824ZM1.51885 6.16985L7.51885 2.52824L6.48115 0.818505L0.481151 4.46012L1.51885 6.16985ZM2 13.1185V5.31499H0V13.1185H2ZM13.6488 22.5887L1.6488 12.3575L0.351203 13.8794L12.3512 24.1106L13.6488 22.5887ZM13.6488 24.1106L25.6488 13.8794L24.3512 12.3575L12.3512 22.5887L13.6488 24.1106ZM26 13.1185V5.31499H24V13.1185H26ZM25.5188 4.46012L19.5189 0.818505L18.4812 2.52824L24.4812 6.16985L25.5188 4.46012ZM18.4812 0.818504L12.4812 4.46012L13.5188 6.16985L19.5189 2.52824L18.4812 0.818504Z"
                      stroke="white"
                      stroke-width="2px"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="sc-f4478397-8 brQQyq"
          />
        </div>
      </a>
    </li>
  </ul>
</div>
`;
